System.IO.FileSystem/PathInternal.cs

```diff
@@ -0,0 +1,273 @@
+            return IsExtended(path)
+                   ||
+                   (
+                       path.Length >= DevicePrefixLength
+                       && IsDirectorySeparator(path[0])
+                       && IsDirectorySeparator(path[1])
+                       && (path[2] == '.' || path[2] == '?')
+                       && IsDirectorySeparator(path[3])
+                   );
+        }
+
+        /// <summary>
+        /// Returns true if the path is a device UNC (\\?\UNC\, \\.\UNC\)
+        /// </summary>
+        internal static bool IsDeviceUNC(string path)
+        {
+            return path.Length >= UncExtendedPrefixLength
+                   && IsDevice(path)
+                   && IsDirectorySeparator(path[7])
+                   && path[4] == 'U'
+                   && path[5] == 'N'
+                   && path[6] == 'C';
+        }
+
+        /// <summary>
+        /// True if the given character is a directory separator.
+        /// </summary>
+        // [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsDirectorySeparator(char c)
+        {
+            return c is DirectorySeparatorChar or AltDirectorySeparatorChar;
+        }
+
+        /// <summary>
+        /// Returns true if the path is effectively empty for the current OS.
+        /// For unix, this is empty or null. For Windows, this is empty, null, or
+        /// just spaces ((char)32).
+        /// </summary>
+        internal static bool IsEffectivelyEmpty(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return true;
+            }
+
+            foreach (var c in path.ToCharArray())
+            {
+                if (c != ' ')
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns true if the path uses the canonical form of extended syntax ("\\?\" or "\??\"). If the
+        /// path matches exactly (cannot use alternate directory separators) Windows will skip normalization
+        /// and path length checks.
+        /// </summary>
+        internal static bool IsExtended(string path)
+        {
+            // While paths like "//?/C:/" will work, they're treated the same as "\\.\" paths.
+            // Skipping of normalization will *only* occur if back slashes ('\') are used.
+            return path.Length >= DevicePrefixLength
+                   && path[0] == '\\'
+                   && (path[1] == '\\' || path[1] == '?')
+                   && path[2] == '?'
+                   && path[3] == '\\';
+        }
+
+        /// <summary>
+        /// Returns true if the given character is a valid drive letter
+        /// </summary>
+        internal static bool IsValidDriveChar(char value)
+        {
+            return (uint)((value | 0x20) - 'a') <= (uint)('z' - 'a');
+        }
+
+        /// <summary>
```

Comment:
are the comments still valid? Especially things like the remarks on the C:\PROGRA~1\A.TXT and other things like this?

Reply:
We "only" have support for UTF-8... Why do we need this?

Reply:
This checks that the character is in the range of `[A-Za-z]` it's used in `IsPathRooted` and `GetRootLength` so that a valid drive root (ex: "C:") passes but and invalid drive root (ex: "_:") does not.